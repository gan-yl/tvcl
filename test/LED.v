// Copyright (c) 2014 Takashi Toyoshima <toyoshim@gmail.com>.
// All rights reserved.  Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

`timescale 100ps/100ps

module LEDTest;
  wire       w_a;
  wire       w_b;
  wire       w_c;
  wire       w_d;
  wire       w_e;
  wire       w_f;
  wire       w_g;
  wire       w_dp;

  reg        clk;
  reg        rst_x;

  always #1 clk = ~clk;

  LED dut(
      .clk  (clk  ),
      .rst_x(rst_x),
      .o_a  (w_a  ),
      .o_b  (w_b  ),
      .o_c  (w_c  ),
      .o_d  (w_d  ),
      .o_e  (w_e  ),
      .o_f  (w_f  ),
      .o_g  (w_g  ),
      .o_dp (w_dp ));

  initial begin
    #18
    forever begin
      #2
      $strobe("%b", w_a,
              "%b", w_b,
              "%b", w_c,
              "%b", w_d,
              "%b", w_e,
              "%b", w_f,
              "%b", w_g,
              "%b", w_dp);
    end
  end

  initial begin
    //$dumpfile("LED.vcd");
    //$dumpvars(0, dut);
    #10
    clk   <= 1'b0;
    rst_x <= 1'b0;
    #10
    rst_x <= 1'b1;
    #100
    $finish;
  end  // initial

endmodule  // LEDTest
