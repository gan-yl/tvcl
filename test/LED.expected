00000010
00000011
10011110
10011111
00100100
00100101
00001100
00001101
10011000
10011001
01001000
01001001
01000000
01000001
00011110
00011111
00000000
00000001
00001000
00001001
00010000
00010001
11000000
11000001
01100010
01100011
10000100
10000101
01100000
01100001
01110000
01110001
00000010
00000011
10011110
10011111
00100100
00100101
00001100
00001101
10011000
10011001
01001000
01001001
01000000
01000001
00011110
00011111
00000000
00000001
00001000
